src/main/java/com/exadel/discount/dto/CityDTO.java

```diff
@@ -0,0 +1,20 @@
+package com.exadel.discount.dto;
+
+import com.exadel.discount.dto.validation.Create;
+import com.exadel.discount.entity.Country;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Null;
+import java.util.UUID;
+
+public class CityDTO {
+    @Null(groups = Create.class)
+    private UUID id;
+
+    @NotBlank(groups = Create.class)
+    private String name;
+
+    @NotNull(groups = Create.class)
+    private Country country;
```

Comment:
I think Country here is a mistake. At least it should be CountryDto. But when you create a new city, maybe it's not a good idea to send full CountryDto, it should be just id

Reply:
> 
> 
> I think Country here is a mistake. At least it should be CountryDto. But when you create a new city, maybe it's not a good idea to send full CountryDto, it should be just id

Done